core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -677,10 +682,35 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
   /**
    * Replay the given log file, saving the application in the listing db.
    */
+
   protected def mergeApplicationListing(
       fileStatus: FileStatus,
       scanTime: Long,
       enableOptimizations: Boolean): Unit = {
+    try {
+      mergeApplicationListingHelper(fileStatus, scanTime, enableOptimizations)
```

Comment:
The name is a bit odd - there're some patterns for this case, like adding `do` as prefix, and more.

Reply:
Thanks, done.